src/julia.h

```diff
@@ -749,7 +749,7 @@ enum jlsmall_typeof_tags {
     jl_bitstags_first = jl_char_tag, // n.b. bool is not considered a bitstype, since it can be compared by pointer
     jl_max_tags = 64
 };
-extern jl_datatype_t *small_typeof[(jl_max_tags << 4) / sizeof(jl_datatype_t*)];
+extern JL_DLLIMPORT jl_datatype_t *small_typeof[(jl_max_tags << 4) / sizeof(jl_datatype_t*)];
```

Comment:
Ah, I see. Yeah, `msvc` is being pedantic, and needs this to be marked `JL_DLLIMPORT` in the foreign library, or it will require it to be found in the current linkage environment and not in any shared library.
```suggestion
#ifndef JL_LIBRARY_EXPORTS
JL_DLLIMPORT
#endif
```